src/main/java/frc/robot/Robot.java

```diff
@@ -28,8 +34,32 @@ public void robotInit() {
     // Instantiate our RobotContainer.  This will perform all our button bindings, and put our
     // autonomous chooser on the dashboard.
     m_robotContainer = new RobotContainer();
+		// register subsystems into a list to minimize redundancy if logging
+		// inputs/outputs for all
+		inst = NetworkTableInstance.create();
+
+		// start a NT4 client
+		inst.startClient4("robot network table client");
+
+		// connect to a roboRIO with team number TEAM
+		inst.setServerTeam(8248);
+
+		// connect to a specific host/port
+		inst.setServer("10.0.0.2", NetworkTableInstance.kDefaultPort4);
+		USB2 = inst.getTable("photonvision").getSubTable("USB2.0_PC_CAMERA");
+		if (OPEN_TAG_WINDOW) {
+			VisionWindow.startWindow(this);
+		}
   }
 
+	public double[] getPose() {
+		if (USB2.getEntry("hasTarget").getBoolean(false)) {
```

Comment:
This is a bit confusing to read... It seems like this translates to something like "if (hasTarget == false) then return a default pose"? This could stand to be broken up a bit and have some comments added to increase clarity.

Also, constants should be in Constants.java; and this default double array you're returning here could probably also be a constant, right?